src/DssSpell.sol

```diff
@@ -23,47 +23,45 @@ import "dss-exec-lib/DssAction.sol";
+    function transferFrom(address, address, uint256) external returns (bool);
 }
 
-interface DssVestLike {
-    function yank(uint256) external;
-    function restrict(uint256) external;
-    function create(
-        address _usr,
-        uint256 _tot,
-        uint256 _bgn,
-        uint256 _tau,
-        uint256 _eta,
-        address _mgr
-  ) external returns (uint256);
+interface L1EscrowLike {
+    function approve(address, address, uint256) external;
 }
 
+
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
     // Hash: seth keccak -- "$(wget https://raw.githubusercontent.com/makerdao/community/3224f50b0b5a9301831213ed858bc1d206de8e40/governance/votes/Executive%20vote%20-%20December%2010%2C%202021.md -q -O - 2>/dev/null)"
     string public constant override description =
-        "2021-12-10 MakerDAO Executive Spell | Hash: 0x9cc240b4c1027d3dc1abb24ecc598703352d8135c8a1670a6591e9a334836e5d";
+        "2022-01-14 MakerDAO Executive Spell | Hash: ";
 
     // --- Math ---
-    uint256 constant MILLION = 10**6;
+    uint256 constant MILLION = 10 ** 6;
+    uint256 constant WAD     = 10 ** 18;
 
     // --- Ilks ---
-    bytes32 constant WSTETH_A = "WSTETH-A";
-    bytes32 constant MATIC_A  = "MATIC-A";
+    bytes32 constant UNIV2WBTCETH_A = "UNIV2WBTCETH-A";
+    bytes32 constant UNIV2UNIETH_A  = "UNIV2UNIETH-A";
+    bytes32 constant UNIV2DAIETH_A  = "UNIV2DAIETH-A";
+    bytes32 constant UNIV2USDCETH_A = "UNIV2USDCETH-A";
+    bytes32 constant UNIV2WBTCDAI_A = "UNIV2WBTCDAI-A";
 
     // --- Wallet addresses ---
-    address constant GRO_WALLET = 0x7800C137A645c07132886539217ce192b9F0528e;
-    address constant ORA_WALLET = 0x2d09B7b95f3F312ba6dDfB77bA6971786c5b50Cf;
-    address constant PE_WALLET  = 0xe2c16c308b843eD02B09156388Cb240cEd58C01c;
-
-    // --- Dates ---
-    uint256 constant MAY_01_2021 = 1619827200;
-    uint256 constant JUN_21_2021 = 1624233600;
-    uint256 constant JUL_01_2021 = 1625097600;
-    uint256 constant SEP_13_2021 = 1631491200;
-    uint256 constant SEP_20_2021 = 1632096000;
+    address constant FLIP_FLOP_FLAP_WALLET  = 0x688d508f3a6B0a377e266405A1583B3316f9A2B3;
+    address constant FEEDBLACK_LOOPS_WALLET = 0x80882f2A36d49fC46C3c654F7f9cB9a2Bf0423e1;
+    address constant SCHUPPI_WALLET         = 0x89C5d54C979f682F40b73a9FC39F338C88B434c6;
+    address constant MAKERMAN_WALLET        = 0x9AC6A6B24bCd789Fa59A175c0514f33255e1e6D0;
+    address constant ACRE_INVEST_WALLET     = 0x5b9C98e8A3D9Db6cd4B4B4C1F92D0A551D06F00D;
+    address constant JUSTIN_CASE_WALLET     = 0xE070c2dCfcf6C6409202A8a210f71D51dbAe9473;
+    address constant GFX_LABS_WALLET        = 0xa6e8772af29b29B9202a073f8E36f447689BEef6;
+
+    // --- Optimism Recovery Mainnet Addresses
+    address immutable MCD_DAI             = DssExecLib.dai();
```

Comment:
Does this work?

Reply:
Specifically, can we assign an immutable outside of a constructor like this? We may have to get these values inside of the action or hardcode them as constants, but I'd be happily surprised if otherwise.

Reply:
I am less familiar with all the rules around the `immutable` stuff since it came out in my gap time... but the tests pass with this this way so it seems like it is working.  I'm happy to change it.  I had originally put it as just the hard coded address because I thought that was our preference for spells (hard code rather than call to something).  i'm open either way (cc: @julienmartinlevrai and @godsflaw )

Reply:
Yes, this way of doing it is also surprising for me. I had never seen it done that way but I like it. As for constants, I think our préférence is to hardcode things *unless* we can get them from DssExecLib.

Reply:
I would prefer not to do "surprising" things in the spell.

So I would 

1) return to hardcoding the values into constants

or

2) Define them as variables inside the action function and call out to dssexeclib (seems like we have done this in past spells)

Do you all have a preference?

Reply:
I would suggest to leave it as is. While we had never done it, it does seem expected behavior as per the [0.6.12 documentation](https://docs.soliditylang.org/en/v0.6.12/contracts.html#immutable):

> Immutable variables can be assigned an arbitrary value in the constructor of the contract or at the point of their declaration.

Reply:
I'm honestly concerned about the execution context being a delegate call here.  It's clearly working though, which means immutable gets tossed into a different memory space.  I tested to see if this would fail if we removed immutable, and it did.

```haskell
$ make test-forge
./test-dssspell-forge.sh match="" block=""
compiling...
success.
Running 14 tests for DssSpellTest.json:DssSpellTest
[FAIL. Reason: ds-pause-delegatecall-error] testCastCost() (gas: 1003540)
[FAIL. Reason: ds-pause-delegatecall-error] testDelegatePayments() (gas: 1014563)
[PASS] testFailTooEarly() (gas: 3647)
[PASS] testFailTooLate() (gas: 3591)
[PASS] testFailWrongDay() (gas: 3592)
[PASS] testFail_notScheduled() (gas: 4725)
[FAIL. Reason: ds-pause-delegatecall-error] testOnTime() (gas: 1003015)
[FAIL. Reason: ds-pause-delegatecall-error] testOptimismRecovery() (gas: 1014266)
[FAIL. Reason: ds-pause-delegatecall-error] testSpellIsCast_GENERAL() (gas: 2548967)
[FAIL. Reason: ds-pause-delegatecall-error] test_auth() (gas: 1003033)
[FAIL. Reason: ds-pause-delegatecall-error] test_auth_in_sources() (gas: 1003078)
[PASS] test_bytecode_matches() (gas: 1539387)
[PASS] test_nextCastTime() (gas: 337535)
[PASS] test_use_eta() (gas: 336462)
Error: 
   0: Encountered a total of 7 failing tests, 7 tests succeeded

Location:
   cli/src/cmd/test.rs:187

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
make: *** [test-forge] Error 1
```

Reply:
Okay, I experimented a little and talked myself into this being okay.

Reply:
I could go either way here. On one hand, since the end result is that this will ultimately be a constant it's cheaper to deploy to just keep this constant, since we would avoid the external call.
On the other hand, this is portable between the Goerli spell and this and doing it this way probably saves as much developer time as we're spending on deployment gas (when we aren't bikeshedding it)

Reply:
Can we at least make it `public` so that we can inspect it on-chain.